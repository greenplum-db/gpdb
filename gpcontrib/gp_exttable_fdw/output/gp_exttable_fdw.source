--
-- Test foreign-data wrapper gp_exttable_fdw.
--
CREATE EXTENSION IF NOT EXISTS gp_exttable_fdw;
NOTICE:  extension "gp_exttable_fdw" already exists, skipping
CREATE SERVER IF NOT EXISTS gp_exttable_server FOREIGN DATA WRAPPER gp_exttable_fdw;
NOTICE:  server "gp_exttable_server" already exists, skipping
-- ===================================================================
-- tests for validator
-- ===================================================================
-- Error, undefined option
CREATE FOREIGN TABLE tableless_ext_fdw(a int, b int)
SERVER gp_exttable_server
OPTIONS (undefined_option ',');
ERROR:  invalid option "undefined_option"  (seg2 127.0.0.1:7004 pid=5174)
HINT:  Valid options in this context are: location_uris, execute_on, command, format_type, reject_limit, reject_limit_type, log_errors, encoding, is_writable, delimiter, null, header, quote, escape, force_not_null, force_quote, fill_missing_fields, newline, quote, format
-- Error, miss format_type option
CREATE FOREIGN TABLE tableless_ext_fdw(a int, b int)
SERVER gp_exttable_server
OPTIONS (delimiter ',',
         location_uris 'file://@hostname@@abs_srcdir@/data/tableless.csv');
ERROR:  must specify format_type option([t | c | b], t(text), c(csv), b(custom))  (seg1 127.0.0.1:7003 pid=5173)
-- Error, invalid format_type
CREATE FOREIGN TABLE tableless_ext_fdw(a int, b int)
SERVER gp_exttable_server
OPTIONS (format_type 'invalid_format_type', delimiter ',',
         location_uris 'file://@hostname@@abs_srcdir@/data/tableless.csv');
ERROR:  format_type must be [t | c | b], t(text), c(csv), b(custom)  (seg2 127.0.0.1:7004 pid=5174)
-- Error, miss both location_uris and command option
CREATE FOREIGN TABLE tableless_ext_fdw(a int, b int)
SERVER gp_exttable_server
OPTIONS (format_type 'c', delimiter ',');
ERROR:  must specify one of locationuris and command option  (seg1 127.0.0.1:7003 pid=5173)
-- Error, conflict location_uris and command option
CREATE FOREIGN TABLE tableless_ext_fdw(a int, b int)
SERVER gp_exttable_server
OPTIONS (format_type 'c', delimiter ',',
         location_uris 'file://@hostname@@abs_srcdir@/data/tableless.csv',
         command 'command');
ERROR:  locationuris and command options conflict with each other  (seg1 127.0.0.1:7003 pid=5173)
-- Error, invalid reject_limit_type option
CREATE FOREIGN TABLE tableless_ext_fdw(a int, b int)
SERVER gp_exttable_server
OPTIONS (format_type 'c', delimiter ',',
         location_uris 'file://@hostname@@abs_srcdir@/data/tableless.csv',
         reject_limit_type 'invalid_reject_limit_type');
ERROR:  reject_limit_type must be [r | p], r(ROW) and p(PERCENT)  (seg1 127.0.0.1:7003 pid=5173)
-- Error, invalid reject_limit option
CREATE FOREIGN TABLE tableless_ext_fdw(a int, b int)
SERVER gp_exttable_server
OPTIONS (format_type 'c', delimiter ',',
         location_uris 'file://@hostname@@abs_srcdir@/data/tableless.csv',
         reject_limit_type 'r', reject_limit '-1');
ERROR:  segment reject limit in ROWS must be 2 or larger (got -1)  (seg1 127.0.0.1:7003 pid=5173)
CREATE FOREIGN TABLE tableless_ext_fdw(a int, b int)
SERVER gp_exttable_server
OPTIONS (format_type 'c', delimiter ',',
         location_uris 'file://@hostname@@abs_srcdir@/data/tableless.csv',
         reject_limit_type 'p', reject_limit '120');
ERROR:  segment reject limit in PERCENT must be between 1 and 100 (got 120)  (seg1 127.0.0.1:7003 pid=5173)
-- OK, no execute_on | log_errors | encoding | is_writable option
CREATE FOREIGN TABLE tableless_ext_fdw(a int, b int)
SERVER gp_exttable_server
OPTIONS (format_type 'c', delimiter ',',
         location_uris 'file://@hostname@@abs_srcdir@/data/tableless.csv');
SELECT * FROM tableless_ext_fdw;
 a | b  
---+----
 1 | 12
 2 | 22
 3 | 32
(3 rows)

