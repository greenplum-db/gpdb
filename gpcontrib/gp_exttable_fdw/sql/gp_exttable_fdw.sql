--
-- Test foreign-data wrapper gp_exttable_fdw.
--

CREATE EXTENSION IF NOT EXISTS gp_exttable_fdw;

CREATE SERVER IF NOT EXISTS gp_exttable_server FOREIGN DATA WRAPPER gp_exttable_fdw;

-- validator tests
CREATE FOREIGN TABLE ext_without_execute_on_option (id int, category text) SERVER gp_exttable_server;  -- ERROR
