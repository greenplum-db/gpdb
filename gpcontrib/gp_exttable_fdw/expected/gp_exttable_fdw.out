--
-- Test foreign-data wrapper gp_exttable_fdw.
--
CREATE EXTENSION IF NOT EXISTS gp_exttable_fdw;
NOTICE:  extension "gp_exttable_fdw" already exists, skipping
CREATE SERVER IF NOT EXISTS gp_exttable_server FOREIGN DATA WRAPPER gp_exttable_fdw;
NOTICE:  server "gp_exttable_server" already exists, skipping
-- validator tests
CREATE FOREIGN TABLE ext_without_execute_on_option (id int, category text) SERVER gp_exttable_server;  -- ERROR
ERROR:  must specify option: execute_on  (seg0 127.0.0.1:7002 pid=8255)
